lib/errors.js

```diff
@@ -185,4 +185,76 @@ errors.GatewayTimeoutError = function(message) {
 };
 util.inherits(errors.GatewayTimeoutError, errors.SmartcarError);
 
+/**
+ * Enhanced errors from API v2.0
+ *
+ * @param {Object|String} error - response body from a v2.0 request
+ */
+errors.SmartcarErrorV2 = class extends errors.SmartcarError {
+  constructor(error) {
+    if (typeof error === 'string') {
+      super(error);
+      this.description = error;
+      this.name = 'SmartcarErrorV2';
+      return;
+    } else {
+      super(error ? error.message : null);
```

Comment:
Shouldn't this be 
```suggestion
      super(`${error.type}:${error.code} - ${error.description`)
```

?